Win installer

tschneidereit opened the pull request.

tschneidereit commented:
(I'll provide a proper description once this actually works.)

kubkon commented:
Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?

tschneidereit commented:
> Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?

These are still statically linked binaries, so no, that's not it.

On Windows there isn't on obvious place to drop binaries into, so you'd commonly create a directory under `Program Files` and, for applications that should be available on the command line, add that to `%PATH%`. That's not really something our users should have to do manually.

To be clear, we should definitely also have a Homebrew recipe, and eventually be in distro package managers and all that :)

kubkon commented:
> > Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?
> 
> These are still statically linked binaries, so no, that's not it.
> 
> On Windows there isn't on obvious place to drop binaries into, so you'd commonly create a directory under `Program Files` and, for applications that should be available on the command line, add that to `%PATH%`. That's not really something our users should have to do manually.
> 

Ah, that's fair enough!

> To be clear, we should definitely also have a Homebrew recipe, and eventually be in distro package managers and all that :)

Yep, that's a must! I'm sure @marmistrz and myself will be able to help out with `pacman` and homebrew respectively :-)

tschneidereit commented:
> Yep, that's a must! I'm sure @marmistrz and myself will be able to help out with `pacman` and homebrew respectively :-)

That'd be fantastic!

alexcrichton commented on .azure-pipelines.yml:
```diff
@@ -156,28 +156,30 @@ jobs:
-        archiveType: 'zip'
-        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-$(tagName)-x86_64-windows.zip'
-      displayName: Archive files (Win)
-      condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
```
I've found that for automated installation of tools it's sometimes nifty to still have a tarball-like installer for Windows that's not just the executable ones. Perhaps both an `*.msi` could be created as well as a `*.tar.xz` to match the other platforms.

(I'm assuming that some sort of installer website would smoth over "what to choose" here as well)

tschneidereit replied:
Good point, yes. I left the zip file in place with the latest version here.

tschneidereit commented:
@alexcrichton this is ready for review again. I added the `LICENSE` and `README.md` files to the installer and moved the executables into a `bin` folder, which seems to be what everyone's doing.

tschneidereit commented:
(Oh, and I had a CI check running in my own repo and stupidly canceled this one. I just triggered [a new one](https://dev.azure.com/CraneStation/Wasmtime/_build/results?buildId=99))
